go/vt/topo/k8stopo/server.go

```diff
@@ -47,16 +48,24 @@ import (
 	vttyped "vitess.io/vitess/go/vt/topo/k8stopo/client/clientset/versioned/typed/topo/v1beta1"
 )
 
-var (
+var kubeconfigPath, configContext, configNamespace string
+var k8stopoFlags *pflag.FlagSet
+
+func init() {
+	k8stopoFlags = pflag.NewFlagSet("k8stopo", pflag.ExitOnError)
+	registerFlags(k8stopoFlags)
+}
```

Comment:
I don't understand this part 😕. Shouldn't we register the k8stopo like the other topo services?

Reply:
The other mechanisms didn't work. Maybe there is a better way, but I needed to do this in `go/vt/topo/k8stopo/server_flaky_test.go`  using the above technique  to fix `TestKubernetesTopo`.

```
k8stopoFlags.Set("topo_k8s_kubeconfig", testConfigPath)
```

Reply:
Can't you directly set `kubeconfigPath = testConfigPath`? I think we do that in other test code too.
We use something like 
```go
oldConfigPath := kubeconfigPath
kubeconfigPath = testPath
defer func() {
    kubeconfigPath = oldConfigPath
}()
```

Reply:
Also, don't we need these flags in the binaries that also have the zookeeper flags, like VTOrc etc?

Reply:
I think they will be used if someone uses `k8s` flavour of topo-server, but I am not sure.

Reply:
imo, this code should happen only in the flaky test file (ie. you should do what things like `tabletconntest.SetProtocol` do)

Reply:
@GuptaManan100's suggestion above worked.